plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,930 @@
+}
+
+// netconfYangPush notification received through NETCONF, as per specification
+type netconfYangPush struct {
+	netconf.Notification
+	PushUpdate struct {
+		Content struct {
+			InnerXML []byte `xml:",innerxml"`
+		} `xml:"datastore-contents-xml"`
+	} `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push push-update"`
+}
+
+// netconfYangEvent notification received through NETCONF, as per specification
+type netconfYangEvent struct {
+	netconf.Notification
+	InnerXML []byte `xml:",innerxml"`
+}
+
+// CiscoTelemetryNETCONF plugin
+type CiscoTelemetryNETCONF struct {
+	ServerAddress string `toml:"server_address"`
+
+	Dsrs *dialinSubscriptionRequestsService `toml:"subscription_service"`
+	Grs  *getRequestsService                `toml:"get_service"`
+
+	// SSH connection - credentials
+	Username string `toml:"username"`
+	Password string `toml:"password"`
+
+	// SSH connection - security
+	IgnoreServerAuthenticity bool   `toml:"ignore_server_authenticity"`
+	ServerPublicKey          string `toml:"server_public_key"`
+
+	// Xpaths and tags of all operations
+	userXpaths map[string]interface{}
+	userTags   map[string]interface{}
+
+	Redial config.Duration `toml:"redial"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	acc    telegraf.Accumulator
+	cancel context.CancelFunc
+
+	// Waitgroup for monitoring the main services
+	waitgroup sync.WaitGroup
+}
+
+// Init ensures the plugin is configured correctly.
+func (c *CiscoTelemetryNETCONF) Init() error {
+	if c.ServerAddress == "" {
+		return fmt.Errorf("server address cannot be empty")
+	}
+	if c.Username == "" || c.Password == "" {
+		return fmt.Errorf("ssh credentials cannot be empty")
+	}
+	if !c.IgnoreServerAuthenticity && c.ServerPublicKey == "" {
+		return fmt.Errorf("public key must exist when ignore server authenticity is disabled")
+	}
+	if time.Duration(c.Redial).Seconds() < 1 {
+		return fmt.Errorf("redial must be greater than or equal to 1s")
+	}
+	return nil
+}
+
+// createService calls functionalities for the dial-in subscriptions service
+func (dsrs *dialinSubscriptionRequestsService) createService(
+	ctx context.Context, c *CiscoTelemetryNETCONF) {
+
+	dsrs.setting = &setting{Client: nil, Session: nil,
+		Mutex: new(sync.Mutex), state: 0}
+
+	defer c.waitgroup.Done()
+
+	for ctx.Err() == nil {
+		c.connectClient(ctx, dsrs.setting)
+		if dsrs.setting.state == connected {
+			dsrs.createRequests(ctx, c)
+		}
+		if dsrs.setting.state == created {
+			// Launch 2 goroutines: one for reception of telemetry data,
+			//  and one for reception of event notifications
+			var waitgroup sync.WaitGroup
+
+			if len(dsrs.Subscriptions) > 0 {
+				waitgroup.Add(1)
+				go func() {
+					dsrs.receiveTelemetry(ctx, c)
+					waitgroup.Done()
+				}()
+			}
+
+			if len(dsrs.Notifications) > 0 {
+				waitgroup.Add(1)
+				go func() {
+					dsrs.receiveNotifications(ctx, c)
+					waitgroup.Done()
+				}()
+			}
+
+			waitgroup.Wait()
+		}
+
+		select {
+		case <-ctx.Done():
```

Comment:
Maybe
```suggestion
		select {
		case <-ctx.Done():
			return
```
and change the for loop to `for {` to directly indicate the infinite loop?